packages/client/src/encryption/GroupKey.ts

```diff
@@ -36,12 +36,48 @@ function GroupKeyObjectFromProps(data: GroupKeyProps | GroupKeyObject): GroupKey
 
 export type GroupKeyish = GroupKey | GroupKeyObject | ConstructorParameters<typeof GroupKey>
 
-// eslint-disable-next-line no-redeclare
+/**
+ * GroupKeys are AES cipher keys, which are used to encrypt/decrypt StreamMessages (when encryptionType is AES).
+ * Each group key contains 256 random bits of key data and an UUID.
+ *
+ * A group key stores the same key data in two fields: the bytes as hex-encoded string, and as a raw Uint8Array.
+ * TODO: If this data duplication doesn't give us any performance improvement we could store the key data only
+ * in one field.
+ */
+
 export class GroupKey {
     /** @internal */
     static InvalidGroupKeyError = InvalidGroupKeyError
 
-    static validate(maybeGroupKey: GroupKey): void | never {
+    /** @internal */
+    id: string
+    /** @internal */
+    hex: string
+    /** @internal */
+    data: Uint8Array
+
+    constructor(groupKeyId: string, groupKeyBufferOrHexString: Uint8Array | string) {
+        this.id = groupKeyId
+        if (!groupKeyId) {
+            throw new InvalidGroupKeyError(`groupKeyId must not be falsey ${inspect(groupKeyId)}`)
+        }
+
+        if (!groupKeyBufferOrHexString) {
+            throw new InvalidGroupKeyError(`groupKeyBufferOrHexString must not be falsey ${inspect(groupKeyBufferOrHexString)}`)
+        }
+
+        if (typeof groupKeyBufferOrHexString === 'string') {
+            this.hex = groupKeyBufferOrHexString
+            this.data = Buffer.from(this.hex, 'hex')
+        } else {
+            this.data = groupKeyBufferOrHexString
+            this.hex = Buffer.from(this.data).toString('hex')
+        }
+
+        (this.constructor as typeof GroupKey).validate(this)
```

Comment:
Regarding original code, some weird indirectness here?
```suggestion
        GroupKey.validate(this)
```